Source/VMware.vSphereDSC/DSCResources/VDSwitchVMHost.ps1

```diff
@@ -0,0 +1,251 @@
+
+            $result.Server = $this.Connection.Name
+            $result.VMHostNames = $vmHosts | Select-Object -ExpandProperty Name
+            $result.Ensure = $this.Ensure
+
+            if ($null -eq $distributedSwitch) {
+                # If the Distributed Switch is $null, it means that Ensure is 'Absent' and the Distributed Switch does not exist.
+                $result.VdsName = $this.Name
+            }
+            else {
+                $result.VdsName = $distributedSwitch.Name
+            }
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Distributed Switch with the specified name from the server if it exists.
+    If the Distributed Switch does not exist and Ensure is set to 'Absent', $null is returned.
+    Otherwise it throws an exception.
+    #>
+    [PSObject] GetDistributedSwitch() {
+        if ($this.Ensure -eq [Ensure]::Absent) {
+            $distributedSwitch = Get-VDSwitch -Server $this.Connection -Name $this.VdsName -ErrorAction SilentlyContinue
+            return $distributedSwitch
+        }
+        else {
+            try {
+                $distributedSwitch = Get-VDSwitch -Server $this.Connection -Name $this.VdsName -ErrorAction Stop
+                return $distributedSwitch
+            }
+            catch {
+                throw "Could not retrieve Distributed Switch $($this.VdsName). For more information: $($_.Exception.Message)"
+            }
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the VMHosts with the specified names from the server if they exist.
+    For every VMHost that does not exist, a warning message is shown to the user without throwing an exception.
+    #>
+    [array] GetVMHosts() {
+        $vmHosts = @()
+
+        foreach ($vmHostName in $this.VMHostNames) {
+            $vmHost = Get-VMHost -Server $this.Connection -Name $vmHostName -ErrorAction SilentlyContinue
+            if ($null -eq $vmHost) {
+                Write-WarningLog -Message "The passed VMHost {0} was not found and it will be ignored." -Arguments @($vmHostName)
+            }
+            else {
+                $vmHosts += $vmHost
+            }
+        }
+
+        return $vmHosts
+    }
+
+    <#
+    .DESCRIPTION
+
+    Checks if VMHosts should be added/removed from the Distributed Switch depending on the value of the Ensure property.
+    If Ensure is set to 'Present', checks if all passed VMHosts are part of the Distributed Switch.
+    If Ensure is set to 'Absent', checks if all passed VMHosts are not part of the Distributed Switch.
+    #>
+    [bool] ShouldUpdateVMHostsInDistributedSwitch($vmHosts, $distributedSwitch) {
+        if ($this.Ensure -eq [Ensure]::Present) {
+            foreach ($vmHost in $vmHosts) {
+                $addedVMHost = $vmHost.ExtensionData.Config.Network.ProxySwitch | Where-Object -FilterScript { $_.DvsName -eq $distributedSwitch.Name }
```

Comment:
Is this the only way to discover whether a host is part of a switch?

Reply:
Another way is:
```
$vmHost = Get-VMHost -Name <VMHost Name>
Get-VDSwitch -Name <VD Switch Name> -VMHost $vmHost
```

But this way we will have two Server calls for each VMHost, so I think the current approach is more optimal.